Add 'delivered' event on PUBACK handler

pkinney opened the pull request:
This addresses https://github.com/mcollina/aedes/issues/10 and emits a `delivered` event when a PUBACK is received.

mcollina commented:
Can you please do the same for PUBCOM and add the doc relative to this in the README?
Also, I will call that `'deliver'` (we are already breaking backward compat with mosca :/).

pkinney commented:
Right now, the `PUBCOMP` case currently falls through to the `PUBACK` case (https://github.com/mcollina/aedes/blob/master/lib/handlers/index.js#L35-L38), so the change to the `handlePuback` function should handle both cases.  I added a test for PUBCOMP to make sure and for regression tests.

I changed the event to 'deliver' and updated the test.  Working on updates to README now.

pkinney commented:
README updated.  Let me know if there is anything else to add.

mcollina commented on test/client-pub-sub.js:
```diff
@@ -73,6 +73,77 @@ test('publish direct to a single client QoS 1', function (t) {
+    client.publish({
+      topic: 'hello',
+      payload: new Buffer('world'),
+      qos: 1
```
This should be qos:2

pkinney replied:
Yep. Corrected.

mcollina commented:
Release as v0.14.0

behrad commented on lib/handlers/puback.js:
```diff
@@ -3,6 +3,7 @@
 function handlePuback (client, packet, done) {
   var persistence = client.broker.persistence
   persistence.outgoingClearMessageId(client, packet, done)
+  client.broker.emit('deliver', packet, client)
```
I believe `delivered` or `ack` could be a better name, in Mosca we are using `delivered` @mcollina

mcollina replied:
I think 'ack' is the best of all, actually. I'll get that updated asap!
Il giorno dom 3 apr 2016 alle 18:14 Behrad <notifications@github.com> ha
scritto:

> In lib/handlers/puback.js
> <https://github.com/mcollina/aedes/pull/39#discussion_r58312605>:
>
> > @@ -3,6 +3,7 @@
> >  function handlePuback (client, packet, done) {
> >    var persistence = client.broker.persistence
> >    persistence.outgoingClearMessageId(client, packet, done)
> > +  client.broker.emit('deliver', packet, client)
>
> I believe delivered or ack could be a better name, in Mosca we are using
> delivered @mcollina <https://github.com/mcollina>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/mcollina/aedes/pull/39/files/7897baca0c9db516cd179f3d64497c9527b84f5a#r58312605>
>
